pkg/report/handler_test.go

```diff
@@ -45,3 +55,75 @@ func TestShowHandler(t *testing.T) {
 			status, http.StatusOK)
 	}
 }
+
+type roundTripFunc func(req *http.Request) *http.Response
+
+func (f roundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
+
+func TestPersistHandler(t *testing.T) {
+	var scenarios = []struct {
+		fakeResponseStatusCode int
+		getLoadTestsFn         func(action k8stesting.Action) (handled bool, ret runtime.Object, err error)
+		expectedStatusCode     int
+	}{
+		{
+			fakeResponseStatusCode: http.StatusOK,
+			getLoadTestsFn: func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+				return true, &loadtestV1.LoadTest{}, nil
+			},
+			expectedStatusCode: http.StatusOK,
+		},
+		{
+			fakeResponseStatusCode: http.StatusOK,
+			getLoadTestsFn: func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+				return true, nil, k8serrors.NewNotFound(loadtestV1.Resource("loadtests"), "loadtest-name")
+			},
+			expectedStatusCode: http.StatusNotFound,
+		},
+		{
+			fakeResponseStatusCode: http.StatusUnauthorized,
+			getLoadTestsFn: func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+				return true, &loadtestV1.LoadTest{}, nil
+			},
+			expectedStatusCode: http.StatusUnauthorized,
+		},
+	}
+
+	req, err := http.NewRequest("PUT", "/load-test/loadtest-name/report", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// init dependencies for report package
+	minioClient, _ = minio.NewWithRegion("localhost:80", "access-key", "secret-access-key", false, "us-east1")
+	bucketName = "bucket-name"
+	expires = time.Second
+
+	for _, scenario := range scenarios {
```

Comment:
It would be better to run each test separately with `t.Run()`, so it will be easier to check if some scenario is failed while `go run`